### ISSUE TITLE
tag more information about parent transactions

### ISSUE THREAD

**Issue by gits7r** [spesmilo/electrum#2158]
Not long ago we've added warnings for unconfirmed transactions such as Low Fee, Unconfirmed parent. These are very very useful and great, but they don't work as expected when a parent contains a mixture between unconfirmed (or low fee) and confirmed inputs. The checks for these should be done for each input, not for the transaction as a whole, can we do this?

Example:
- Alice receives in Electrum tx A which spends an unconfirmed input. She sees: Unconfirmed parent-  (good)

- Alice receives in Electrum tx B which spends a confirmed input but has a very low fee, and might get discarded from mempools and never mined. She sees: Low fee - (good).

- Alice receives in Electrum tx C which spends 2 inputs: one unconfirmed, and one confirmed. She sees just Unconfirmed - (Not good, should include info about the unconfirmed input of the parent as well)

- Alice receives in Electrum tx D which spends 2 inputs: one that is unconfirmed and has a very low fee and one confirmed. She sees just Unconfirmed - (Not good should include info about the low fee input of the parent as well).

Just that one input can invalidate the entire tx, thus cancelling everything, or keeping it unconfirmed for very very long time.

**Comment by ecdsa** [comment]
in cases 3 and 4, the expected behaviour is to display "unconfirmed parent". 
(the low fee of the parent in 4 is out of scope)
which electrum server were you connected to?

**Comment by ecdsa** [comment]
@kyuupichan can you check if electrumX handles this correctly?

**Comment by kyuupichan** [comment]
Pretty sure it does; and it's dynamic so is recalculated with each request.  It's possible the person connected to a server that had just started and so didn't have the full mempool.

**Comment by ecdsa** [comment]
would it be possible to have the server wait until it has the full mempool, before it starts serving requests?

**Comment by gits7r** [comment]
Just tested - while I can't recall what server I was connected to when experienced the missing info (it was on auto-connect) I now tried to spend one confirmed input and one unconfirmed one in a single transaction using my own server, which wasn't yet upgraded to ElectrumX it's running the latest Electrum-server and pretty sure it has a complete mempool because it's running for like 20 days with no interruptions and the transaction was tagged as expected in wallet, as "Unconfirmed Parent".

Either it was something wrong with the server's code used at that time either @kyuupichan point is right. I see the utility in requesting full mempool before starting to serve requests, it can be done with no down sides.

**Comment by kyuupichan** [comment]
I'm fixing this properly in ElectrumX today:

   https://github.com/kyuupichan/electrumx/issues/129

However in doing so I noticed a bug in ElectrumX which meant that all parents were *always* reported as confirmed.  Hence you cannot have been connected to an ElectrumX server and the issues you saw were from jelectrum or electrum-server servers.

**Comment by gits7r** [comment]
Neat. Yes, my case could have been a non ElectrumX server, but luckily this lead to the fix in ElectrumX since that is the recommended server version.
